docs/docs/concepts/flows.mdx

```diff
@@ -483,6 +483,44 @@ flow is completed.
   scope: "flow"
 ```
 
+#### Slot value validation
+
+You can now write slot value validation directly in your flow yaml file without the need to use custom actions.
+Add a `rejections` property to your `collect_information` step to define the validation rules.
+The `rejections` section is a list of mappings, where each mapping has an `if` and an `utter` property:
+- the `if` property is a condition written in natural language that is evaluated using the
+[pypred](https://github.com/armon/pypred) library.
+- the `utter` property is the name of the response the assistant will send if the condition evaluates to `True`.
+
+Here is an example:
+
+```yaml
+flows:
+  verify_eligibility:
+    description: This flow verifies if the user is eligible for creating an account.
+    name: verify eligibility
+    steps:
+      - id: "ask_age"
+        collect: "age"
+        rejections:
+          - if: age < 1
+            utter: utter_invalid_age
+          - if: age < 13
+            utter: utter_age_at_least_13
+        next: "ask_email"
+```
+
+Upon every validation failure, the assistant will again make another attempt to collect the information.
+This will be repeated until none of the rejection conditions pass. Once all rejections have evaluated to `False`, the
+assistant moves on to the `next` step with the collected value.
+
+If the predicate defined in the `if` property is invalid or missing, the assistant will log an error and respond with the
+`utter_internal_error_rasa` response which by default is
+`Sorry, I'm having trouble understanding you right now. Please try again later.`. However, you can customize this in
+your domain file. We recommend using end-to-end testing to ensure that your flows work as expected.
```

Comment:
usually we give an example of how this customisation looks like. Wondering if it makes sense here or if it makes the page look too complicated. Thoughts?

Reply:
Hmm, tbh I should try it out first as well, haven't yet tested adding this response in the domain file with a different text. Not sure what happens when the default flows are merged (that's where this response resides in OSS).

Reply:
Confirming that I tested the customization of the `utter_internal_error_rasa` and it worked fine, the overridden new response is displayed in the chat. There's a UserWarning displayed during training that duplicate responses were found for `utter_internal_error_rasa`, but training completes successfully.
I think adding an example for overriding the response would clutter the section, i can just say add a response `utter_internal_error_rasa` to your domain file with your custom text.